### PR TITLE
DE461108 : Trusted cert id issue when importing the bundle to target gateway

### PR DESCRIPTION
**Pull request by RaghuReddy**
Issue: Policies are referring source GW certificate ID's in policy XML.

Fixed the issue by leveraging annotation (@bundle-entity) support.

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/TrustedCert.java [hunk 1]
```diff
@@ -207,4 +227,22 @@ public void postLoad(String entityKey, Bundle bundle, @Nullable File rootFolder,
         super.postLoad(entityKey, bundle, rootFolder, idGenerator);
         setName(entityKey);
     }
+
+    public String getType() {
+        return "trusted_cert";
```
Comment (nikhilahuja) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/TrustedCert.java:232]: Use `EntityTypes` instead

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/TrustedCertEntityBuilder.java [hunk 2]
```diff
@@ -72,7 +71,9 @@
             case DEPLOYMENT:
                 return entities.entrySet().stream()
                         .map(
-                                trustedCertEntry -> EntityBuilderHelper.getEntityWithOnlyMapping(TRUSTED_CERT_TYPE, trustedCertEntry.getKey(), idGenerator.generate())
+                                trustedCertEntry -> EntityBuilderHelper.getEntityWithOnlyMapping(TRUSTED_CERT_TYPE, trustedCertEntry.getKey(),
+                                        ((TrustedCert)trustedCertEntry.getValue()).getAnnotatedEntity() !=null && ((TrustedCert)trustedCertEntry.getValue()).getAnnotatedEntity().getId() != null ?
```
Comment (nikhilahuja) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/TrustedCertEntityBuilder.java:75]: Can you please split this into multiple lines and make it more readable

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/TrustedCertEntityBuilder.java [hunk 3]
```diff
@@ -84,7 +85,7 @@
     }
 
     private Entity buildTrustedCertEntity(String name, TrustedCert trustedCert, Map<String, SupplierWithIO<InputStream>> certificateFiles, Document document) {
-        final String id = idGenerator.generate();
+        final String id = trustedCert.getAnnotatedEntity() != null && trustedCert.getAnnotatedEntity().getId() != null ? trustedCert.getAnnotatedEntity().getId(): idGenerator.generate();
```
Comment (nikhilahuja) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/TrustedCertEntityBuilder.java:88]: Please split into `if-else` to make it more readable

**Comment by RaghuReddy** [comment]
addressed in PR https://github.com/CAAPIM/gateway-developer-plugin/pull/252

**Pull request closed without merging**